### ISSUE TITLE
Docker Desktop Uninstall Error: settings-store.json file not found

### ISSUE THREAD

**Issue by a-phe-lei-a** [docker/for-mac#7503]
### Description

When trying to uninstall Docker Desktop on macOS, I'm encountering the following error:

```
/Applications/Docker.app/Contents/MacOS/uninstall
Uninstalling Docker Desktop...
Error: open /Users/a-phe-lei-a/Library/Group Containers/group.com.docker/settings-store.json: no such file or directory
```

Environment:
OS: macOS
Docker Desktop: 15.1.1
Uninstall method: Using the official uninstaller script

Questions:

Is there a way to force the uninstallation despite the missing file?
Should I manually create an empty settings-store.json file before running the uninstaller?

Any help or guidance would be appreciated. Thank you!

### Reproduce

Steps to reproduce:

Ran the uninstaller at /Applications/Docker.app/Contents/MacOS/uninstall
The uninstaller fails with the above error

### docker version

```bash
Client:
 Version:           27.3.1
 API version:       1.47
 Go version:        go1.22.7
 Git commit:        ce12230
 Built:             Fri Sep 20 11:38:18 2024
 OS/Arch:           darwin/arm64
 Context:           default
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

### docker info

```bash
Client:
 Version:    27.3.1
 Context:    default
 Debug Mode: false

Server:
ERROR: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
errors pretty printing info
```

### Diagnostics ID

7454744F-E950-46D2-8D17-87D75F69D321/20241211163957

**Comment by thuyvy3110** [comment]
there's a potential issue with this command: The use of "~/" in quotes won't work as expected. The tilde (~) won't be expanded to your home directory when it's inside quotes.
A corrected version would be:

`echo '{"dataFolder": "/Users/a-phe-lei-a/Library/Containers/com.docker.docker/Data/vms/0/data"}' > ~/Library/Group\ Containers/group.com.docker/settings-store.json`

**Comment by sevagh** [comment]
These instructions are subpar, especially coming from an employee of Docker. Why hardcode "a-phe-lei-a", a name that is meaningless outside of the original reporter of the issue?

**Comment by abgregs** [comment]
I followed these steps. After adding a `settings-store.json` file as directed the  uninstall command does run. But it just hangs forever. Is it just me?

```
Uninstalling Docker Desktop...

```

**Comment by crystalbit** [comment]
I have same. Got stuck with docker. It either doesn't work and doesn't let me uninstall it :)

**Comment by abgregs** [comment]
> I have same. Got stuck with docker. It either doesn't work and doesn't let me uninstall it :)

Yeah, after a force uninstall and complete removal of Docker, a new download and install didn't work for me either.

Could not get the Docker setup dialog or app itself to open and had to repeat force uninstall.

I ended up installing with Homebrew and it finally worked.

**Comment by gieses** [comment]
Using the the build-in terminal from the docker application resolved the issue for me. So the initial `/Applications/Docker.app/Contents/MacOS/uninstall` was giving me problems on the mac terminal but that worked fine in the built-in terminal.
